src/main.rs

```diff
@@ -291,17 +297,36 @@ impl Application {
         let s = self.audio_sink.lock().unwrap();
         let config = s.config.clone();
         let audio_sink = self.audio_sink.clone();
-        let stream = s.device.build_output_stream(
-            &config.into(),
-            move |data: &mut [i16], info: &cpal::OutputCallbackInfo| {
-                let mut audio_sink = audio_sink.lock().unwrap();
-                audio_sink.fill_sound_buffer(data, info);
+        let stream = match s.config.sample_format() {
```

Comment:
Dunno if there's a prettier/cleaner way to do this...

Reply:
This isn't just good, it's good enough :).